src/providers/database.ts

```diff
@@ -206,11 +206,10 @@ export class DeltaSnapshot implements firebase.database.DataSnapshot {
     return valAt(this._delta, this._childPath) !== undefined;
   }
 
-  // TODO(inlined) what is this boolean for?
   forEach(action: (a: DeltaSnapshot) => boolean): boolean {
     let val = this.val();
     if (_.isPlainObject(val)) {
-      _.keys(val).forEach(key => action(this.child(key)));
+      return _.keys(val).some(key => action(this.child(key)));
```

Comment:
Lodash has a helper [`_.some()`](https://lodash.com/docs/4.17.4#some) which we can use directly, instead of having to do `_.keys(val).some()`.